lib/vagrant-cucumber/glue.rb

```diff
@@ -137,6 +139,40 @@ def execute_on_vm ( command, machine, opts = {} )
 
                     end
 
+                    @last_shell_command_output
+
+                end
+
+                def hack_etc_hosts
+
+                    hosts_array = []
+
+                    @vagrant_env.active_machines.each do |name,provider|
+
+                        host_metadata = {}
+                        machine = @vagrant_env.machine(name, provider)
+                        ip = execute_on_vm('facter ipaddress_eth1', machine)
```

Comment:
This assumes facter is installed on the VM in question, which may not always be the case - may be better to implement a shell specific method rather.